p2p/simulations/p2ptest.go

```diff
@@ -19,26 +19,33 @@ import (
 type myProtocol struct {
 	p2p.Protocol
 	proto int
+	peers map[*p2p.Peer]bool
 }
 
 func (p *myProtocol) Run() {
 	fmt.Println("myProtocol Running...", p.proto)
+	p.peers = make(map[*p2p.Peer]bool)
 	//	var peer *p2p.Peer
 	//	var message *p2p.Message
 	ping := time.NewTimer(5 * time.Second)
-loop:
+
 	for {
 		select {
 		case peer := <-p.AddPeerCh:
-			fmt.Println("myProtocol new peer", peer)
+
+			p.peers[peer] = true
+			fmt.Println("myProtocol add new peer. peers=", len(p.peers))
 		case peer := <-p.DelPeerCh:
-			fmt.Println("myProtocol del peer", peer)
+			fmt.Println("myProtocol del peer")
+			// need del from peers
+			delete(p.peers, peer)
```

Comment:
what happened when the peer already been deleted?

Reply:
Peer will send msg to DelPeerCh, subprotocol should handle it.